### ISSUE TITLE
What happened to the idle state in home assistant?

### ISSUE THREAD

**Issue by aleszczynskig** [postlund/pyatv#1672]
### What do you need help with?

Hi,

Previously my Apple TV reported a state of idle when in menu screens (and when showing skip intro buttons in apps) in Home Assistant. With the latest version in HA 2022.02 there are only two states, playing and paused. Has the idle state been removed? I previously used the idle state in automations to detect between a genuine pause which would raise the lights after 5 seconds and a menu or skip button which would raise the lights after 30 seconds. I can no longer make this distinction and am hoping this is a bug or there is an alternative way I can detect the difference.

Thanks for any help in advance.

**Comment by stephenmetzger** [comment]
I'm having the same issue. Not sure when it started occurring.

**Comment by abuchanan920** [comment]
Also interested in this. I have the same issue. I was using a check for having gone to idle for at least 11 seconds (the time those annoying intro buttons stay up) before raising the lights, where as pause would turn them up right away. Now I can't find another way to distinguish and I have to either live with the lights bouncing up and down (and not always successfully at that), or waiting 11 seconds after I paused for the lights to come back on - which is quite a long time to wait actually...

**Comment by abuchanan920** [comment]
Looking into the code - it seems this is directly reported from the AppleTV via the MRP protocol. Previously there was some unknown state that was mapping to 'idle' and now they seem to be reporting 'paused' when those buttons are up. Presumably this started happening in some recent-ish TvOS release.

Probably the two places to look in order to see what I mean:
The MRP protobuf definition: https://github.com/postlund/pyatv/blob/671c24a0b98346e5605e86d080af24a442d485ea/pyatv/protocols/mrp/protobuf/Common.proto#L70
and the logic for using it is at https://github.com/postlund/pyatv/blob/e8ae99be766370ef36543210105abe8c2dbcf611/pyatv/protocols/mrp/player_state.py

Short version is that the chances of being able to do something in pyatv to fix it seem grim...

**Comment by abuchanan920** [comment]
Interesting I still the old behavior on HD Apple TV (model A1625) running the latest tvOS 15.4.1.   The TV where it no longer reports as idle is a brand new 4K Gen2 (also running latest tvOS). So this looks like it make be model dependent behavior (at least in the current os version)

**Comment by postlund** [comment]
@aleszczynskig Can you test with latest HA release? A lot has happened some you opened the issue, so want to verify problem still exists.

**Comment by aleszczynskig** [comment]
I've done some testing and the idle status seems to be working as before, i.e. when in menus or the home screens, at least in terms of media playback. So this issue seems to be resolved.

One small difference I have noted, though I have only tested this with the TV+ app is that when the apps are displaying buttons, like Skip Recap or Skip that the status remains as playing in HA and doesn't change to idle like it did previously. This is actually better in my opinion. I appreciate this is likely due to changes by Apple and pyatv. 

I can now reimplement my automation to turn on the lights immediately when pausing as the status correctly displays as paused.

I believe this issue is fixed so I'll close it.
